src/Agent.Sdk/MountVolume.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Agent.Sdk
+{
+    public class MountVolume
+    {
+        public MountVolume()
+        {
+
+        }
+
+        public MountVolume(string sourceVolumePath, string targetVolumePath, bool readOnly = false)
+        {
+            this.SourceVolumePath = sourceVolumePath;
+            this.TargetVolumePath = targetVolumePath;
+            this.ReadOnly = readOnly;
+        }
+
+        public MountVolume(string fromString)
+        {
+            ParseVolumeString(fromString);
+        }
+
+        private static Regex autoEscapeWindowsDriveRegex = new Regex(@"(^|:)([a-zA-Z]):(\\|/)", RegexOptions.Compiled);
+        private string AutoEscapeWindowsDriveInPath(string path)
+        {
+
+            return autoEscapeWindowsDriveRegex.Replace(path, @"$1$2\:$3");
+        }
+
+        private void ParseVolumeString(string volume)
+        {
+            ReadOnly = false;
+            SourceVolumePath = null;
+
+            string readonlyToken = ":ro";
+            if (volume.ToLower().EndsWith(readonlyToken))
```

Comment:
Nit: can we call `ToLower()` once at the start?

Reply:
we could if we saved it off into a separate variable since the path part of this string needs to be case-sensitive